include/library/spdm_common_lib.h

```diff
@@ -232,16 +232,38 @@ void libspdm_set_last_spdm_error_struct(void *spdm_context,
                                         libspdm_error_struct_t *last_spdm_error);
 
 /**
- * Initialize an SPDM context.
- *
- * The size in bytes of the spdm_context can be returned by libspdm_get_context_size.
+ * Initialize an SPDM context, as well as all secured message contexts,
+ * in the specified locations.
+ *
+ * The size in bytes of the spdm_context can be returned by
+ * libspdm_get_context_size_without_secured_context.
+ * 
+ * The size in bytes of the secured message context can be returned by
+ * libspdm_secured_message_get_total_context_size.
+ * 
+ * @param  spdm_context              A pointer to the SPDM context.
+ * @param  secured_message_contexts  A pointer to the secured message context buffer.
+ *                                   This will hold all secured message contexts.
+ *
+ * @retval RETURN_SUCCESS       Contexts are initialized.
+ * @retval RETURN_DEVICE_ERROR  Context initialization failed.
+ */
+libspdm_return_t libspdm_init_context_with_secure_data_location(void *context,
+    void *secured_message_contexts);
```

Comment:
Open to a name change (for parameters and APIs) if we think secured_message_contexts exposes too much information about internal implementation.

Reply:
I suggest `libspdm_init_context_with_secured_context`. Detail feedback at https://github.com/DMTF/libspdm/pull/1400#issuecomment-1321018164